src/utilities/util_dump.c

```diff
@@ -101,6 +105,18 @@ util_dump(WT_SESSION *session, int argc, char *argv[])
             goto err;
 
         WT_ERR(__wt_buf_set(session_impl, tmp, "", 0));
+        if (timestamp != NULL) {
+            WT_ERR(time_pair_to_timestamp(session_impl, timestamp, tmp));
+            WT_ERR(__wt_buf_catfmt(session_impl, tmp, "isolation=snapshot,"));
+            if ((ret = session->begin_transaction(session, (char *)tmp->data)) != 0) {
+                fprintf(stderr, "%s: begin transaction failed: %s\n", progname,
+                  session->strerror(session, ret));
+                goto err;
+            }
+            __wt_scr_free(session_impl, &tmp);
+            WT_ERR(__wt_scr_alloc(session_impl, 0, &tmp));
+            WT_ERR(__wt_buf_set(session_impl, tmp, "", 0));
+        }
```

Comment:
I think you don't need to re-alloc the `tmp` buffer each time, just re-initialize it. Something like this (entirely untested):
```
*** util_dump.c Thu Feb  6 15:39:19 2020
--- -   Thu Feb  6 15:42:53 2020
***************  
*** 104,111 **** 
          if ((uri = util_uri(session, argv[i], "table")) == NULL)
              goto err;
              
-         WT_ERR(__wt_buf_set(session_impl, tmp, "", 0));
          if (timestamp != NULL) {
              WT_ERR(time_pair_to_timestamp(session_impl, timestamp, tmp));
              WT_ERR(__wt_buf_catfmt(session_impl, tmp, "isolation=snapshot,"));
              if ((ret = session->begin_transaction(session, (char *)tmp->data)) != 0) {
--- 104,111 ---- 
          if ((uri = util_uri(session, argv[i], "table")) == NULL)
              goto err;
              
          if (timestamp != NULL) {
+           WT_ERR(__wt_buf_set(session_impl, tmp, "", 0));
              WT_ERR(time_pair_to_timestamp(session_impl, timestamp, tmp));
              WT_ERR(__wt_buf_catfmt(session_impl, tmp, "isolation=snapshot,"));
              if ((ret = session->begin_transaction(session, (char *)tmp->data)) != 0) {
***************
*** 113,122 ****
                    session->strerror(session, ret));
                  goto err;
              }   
-             __wt_scr_free(session_impl, &tmp);
-             WT_ERR(__wt_scr_alloc(session_impl, 0, &tmp));
-             WT_ERR(__wt_buf_set(session_impl, tmp, "", 0));
          }   
          if (checkpoint != NULL)
              WT_ERR(__wt_buf_catfmt(session_impl, tmp, "checkpoint=%s,", checkpoint));
          WT_ERR(
--- 113,120 ----
                    session->strerror(session, ret));
                  goto err;
              }
          }
+       WT_ERR(__wt_buf_set(session_impl, tmp, "", 0));
          if (checkpoint != NULL)
              WT_ERR(__wt_buf_catfmt(session_impl, tmp, "checkpoint=%s,", checkpoint));
          WT_ERR(
```